lib/pure/httpclient.nim

```diff
@@ -1166,9 +1167,15 @@ proc request*(client: HttpClient | AsyncHttpClient, url: string,
   var lastURL = url
   for i in 1..client.maxRedirects:
     if result.status.redirection():
+      # we get all sorts of troubles if we don't close the connection
+      # before following a redirect. this is inefficient but i didn't
+      # find a better solution.
+      client.close()
```

Comment:
Sorry, but this is far too inefficient. This needs to be fixed in a different way.

Reply:
ok, i am trying another solution. please review ...